Assets/Scripts/Collectables/TimeBasedDropper.cs

```diff
@@ -0,0 +1,52 @@
+﻿using UnityEngine;
+
+namespace Collectables
+{
+    /// <summary>
+    /// A dropper that drops drops based on time.
+    /// </summary>
+    public class TimeBasedDropper : Dropper
+    {
+        /// <summary>
+        /// The frequency with which to drop drops.
+        /// </summary>
+        public int frequency;
+
+        /// <summary>
+        /// Whether this dropper should only drop one drop and then destroy itself.
+        /// </summary>
+        public bool oneTimeDrop;
+
+        private float timeCounter;
+
+        private void Start()
+        {
+            // If the frequency is invalid, destroy the dropper.
+            if (frequency <= 0)
+            {
+                Destroy(this);
+            }
+        }
+
+        private void Update()
+        {
+            if (timeCounter >= frequency)
+            {
+                Transform currentTransform = transform;
+                Instantiate(dropPrefab, currentTransform.position, currentTransform.rotation);
+                if (!oneTimeDrop)
+                {
+                    timeCounter = 0;
+                }
+                else
+                {
+                    Destroy(this);
```

Comment:
I think this should be  `Destroy(this.gameObject)` to delete the entire instance, not just the script or did you want to delete only the script?

Reply:
No, I just want to destroy the dropper, since the enemy could live on and still shoot or move around even if they're done dropping things.